frontends/concrete-python/concrete/fhe/compilation/circuit.py

```diff
@@ -158,58 +106,21 @@ def simulate(self, *args: Any) -> Any:
             Any:
                 result of the simulation
         """
-
-        if not hasattr(self, "simulator"):  # pragma: no cover
-            self.enable_fhe_simulation()
-
-        ordered_validated_args = validate_input_args(self.simulator.client_specs, *args)
-
-        exporter = SimulatedValueExporter.new(self.simulator.client_specs.client_parameters)
-        exported = [
-            (
-                None
-                if arg is None
-                else Value(
-                    exporter.export_tensor(position, arg.flatten().tolist(), list(arg.shape))
-                    if isinstance(arg, np.ndarray) and arg.shape != ()
-                    else exporter.export_scalar(position, int(arg))
-                )
-            )
-            for position, arg in enumerate(ordered_validated_args)
-        ]
-
-        results = self.simulator.run(*exported)
-        if not isinstance(results, tuple):
-            results = (results,)
-
-        decrypter = SimulatedValueDecrypter.new(self.simulator.client_specs.client_parameters)
-        decrypted = tuple(
-            decrypter.decrypt(position, result.inner) for position, result in enumerate(results)
-        )
-
-        return decrypted if len(decrypted) != 1 else decrypted[0]
+        return self._function.simulate(*args)
 
     @property
     def keys(self) -> Keys:
         """
         Get the keys of the circuit.
         """
-
```

Comment:
(todo review) check are keys generated ?

Reply:
ok done tanks lazy access to execution_runtime of the module right?

Reply:
Yes, exactly.